### ISSUE TITLE
Vs Code Extension Error Initializing

### ISSUE THREAD

**Issue by salesforceProDev** [forcedotcom/salesforcedx-vscode#5069]
### Summary

On VScode loading getting this error all the time: Request initialize failed with message: Cannot read properties of undefined (reading 'location') (Lightning Web Components Extension)

### Steps To Reproduce:

1.  Open VsCode.
2.  Let the extensions initialize
3. There is error on the bottom right corner "Request initialize failed with message: Cannot read properties of undefined (reading 'location')"

### Expected result

There should be no error the vsCode extensions should initialize without any error

### Actual result

Getting error: Request initialize failed with message: Cannot read properties of undefined (reading 'location')

### Additional information

Tried to uninstall and install all the extensions back, but didn't work.

<img width="914" alt="Screenshot 2023-09-02 110217" src="https://github.com/forcedotcom/salesforcedx-vscode/assets/97041247/8a57e7a5-a235-4b28-98b9-a3fb5ef1898f">

**Salesforce Extension Version in VS Code**:
 > salesforce.salesforcedx-vscode-lwc --> version: v58.14.1

**SFDX CLI Version**:
sfdx-cli/7.209.6 win32-x64 node-v18.15.0

**OS and version**:
Window 11 64bit

**Comment by JsalvaJ** [comment]
I have the same error. I didn't have it 4 weeks ago. I tried to reinstall the extension package too. 
I tried to download the code again and it didn't work.

![image](https://github.com/forcedotcom/salesforcedx-vscode/assets/39059753/1896e611-8dd2-4060-bc60-a28ea8fba418)

**Comment by CristiCanizales** [comment]
Hello @salesforceDev-role, @JsalvaJ! Are you also getting this error with v58.14.2? We released it 2 days ago, so can you try updating and check if it is still happening?

**Comment by KevinLaurence** [comment]
Hi @CristiCanizales I am also experiencing this issue with version 58.14.2. The extension cannot be updated, and uninstalling and then reinstalling it also does not work. The error message in the log is  "ERR CorruptZip: end of central directory record signature not found".

**Comment by CristiCanizales** [comment]
Hi @salesforceDev-role, @KevinLaurence, @JsalvaJ

If you try installing the vsixes at the bottom of [this page](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/6126453204#:~:text=Size-,58.14.2,-173%20MB), do you still get the same error?

To install just go to extensions view, click the 3 dots at the top right of the sidebar, click install from vsixes and select all the files in the zip (previously unzipped):
<img width="421" alt="image" src="https://github.com/forcedotcom/salesforcedx-vscode/assets/113132642/e9e746ed-0114-40a7-8056-c87148ea1f0c">

**Comment by CristiCanizales** [comment]
If you downgrade to previous versions, like v58.13.1 or v58.11.0, do you still get the same error?

**Comment by salesforceDev-role** [comment]
@CristiCanizales , 

1. The vsixes option you mentioned that is not present in windows 11 vscode. Please find the screenshot attached for your reference.

![LWC Extension VsCode Vsixes](https://github.com/forcedotcom/salesforcedx-vscode/assets/97041247/e051a2d9-be69-423d-adda-bbf6e5a5abb7)

2. I've tried downgrading the version of the extension as well to v58.11.0, no change the same error message is popping up my screen. Attaching the screenshot for your reference.

<img width="734" alt="LWC Extension VsCode Lower Version" src="https://github.com/forcedotcom/salesforcedx-vscode/assets/97041247/535747b7-542e-45e8-9ea0-c6470ac74d63">

**Comment by CristiCanizales** [comment]
@salesforceDev-role I see.. I believe it's something with your system then, because it's happening with every version, even v58.11.0 which is 3 releases previous to the creation of this issue.

Could you tell me your VS Code version? and also maybe try reinstalling/updating VS Code?

**Comment by salesforceDev-role** [comment]
@CristiCanizales, Oh ok I'll try to reinstall it, will update you.

Please find the below versions

VSCode : Version: 1.82.1 (user setup)
Electron: 25.8.0
ElectronBuildId: 23503258
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.22621

**Comment by salesforceDev-role** [comment]
@CristiCanizales , reinstalling is not resolving the issue.

Lets assume if this is a system specific issue, but then why does everyone else also have the same concern?

**Comment by CristiCanizales** [comment]
@salesforceDev-role I know...😓 It's just weird that before v58.14.1 you all were not having issues with previous versions and now you do, and those vsixes for the previous versions didn't change at all... nor anything for the lwc extension

@JsalvaJ, @KevinLaurence would you please tell me your system info too? OS, vscode version, etc

**Comment by KevinLaurence** [comment]
@CristiCanizales My system info is as follows:

Version: 1.82.1 (user setup)
Commit: 6509174151d557a81c9d0b5f8a5a1e9274db5585
Date: 2023-09-08T08:45:05.575Z
Electron: 25.8.0
ElectronBuildId: 23503258
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.19044

In the last 24 hours I have uninstalled all the Salesforce VS Code extensions and reinstalled an earlier version of VS Code. Then attempted to install the Salesforce Extension Pack, which failed.

I then let VS Code upgrade to the latest version (1.82.1) and again tried to install the SF Extension Pack which failed. Then I installed each extension separately and they all installed successfully, with the exception of the LWC extension again.

So, it seems to me that there is something different about salesforce.salesforcedx-vscode-lwc-58.14.2.vsix compared to the other extension files.

**Comment by CristiCanizales** [comment]
@KevinLaurence just to confirm, do you have ESLint extension installed and enabled? It's a dependency for LWC extension

**Comment by JsalvaJ** [comment]
Hello @cristiCanizales, here is my system info.
![image](https://github.com/forcedotcom/salesforcedx-vscode/assets/39059753/3a494cab-4b4b-47e7-be17-f2b5be5be7e1)

**Comment by JsalvaJ** [comment]
I tried to uninstall and reinstall all the things but the error is still ocurring.

**Comment by KevinLaurence** [comment]
> @KevinLaurence just to confirm, do you have ESLint extension installed and enabled? It's a dependency for LWC extension

@CristiCanizales Yes, I have [version 2.4.2](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) installed and it is enabled.

**Comment by KevinLaurence** [comment]
@CristiCanizales I have at least one other colleague who has the same problem with the Lightning Web Components extension and he is running version 58.11.0.

**Comment by gianlucaghezzi24** [comment]
I all I have the same problem with the Lightning Web Components extension version v58.14.2 having also the ESLint extension installed version v2.4.2

**Comment by salesforceDev-role** [comment]
@CristiCanizales what should be done now for the resolution of the same?
Even the same issue I and all my colleagues are facing in their work systems

**Comment by SalesforceAlexS** [comment]
Same issue
https://github.com/forcedotcom/salesforcedx-vscode/issues/5099

**Comment by rob-baillie-ortoo** [comment]
Also experiencing this on macOs

**Comment by CristiCanizales** [comment]
Hello everyone. We're currently digging into this, it's hard because any of us are facing the issue thus we haven't been able to reproduce it. We'll let you know when we have updates! Sorry for the inconveniences

**Comment by bobalicious** [comment]
Have upgraded Java from:
* temurin-11
* jdk-17.0.8
Issue exists with both.

Error in Extensions Host is:

```
2023-09-20 08:40:23.500 [error] Error: Request initialize failed with message: Cannot read properties of undefined (reading 'location')
    at be (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/dist/index.js:23:227)
    at Te (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/dist/index.js:22:12217)
    at Immediate.<anonymous> (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/dist/index.js:22:12068)
    at processImmediate (node:internal/timers:476:21)
```

**Comment by bobalicious** [comment]
Full extensions host output - in case it sheds any light:

```
2023-09-20 08:40:10.335 [info] Extension host with pid 55666 started
2023-09-20 08:40:10.335 [info] Skipping acquiring lock for /Users/xxxx/Library/Application Support/Code/User/workspaceStorage/0801f7ee385f2b392e3a2db3a4de4422.
2023-09-20 08:40:10.501 [info] ExtensionService#_doActivateExtension salesforce.salesforcedx-vscode-apex, startup: false, activationEvent: 'onLanguage:apex-anon'
2023-09-20 08:40:11.922 [warning] DiagnosticCollection with name 'apex-errors' does already exist.
2023-09-20 08:40:11.922 [warning] DiagnosticCollection with name 'apex-errors' does already exist.
2023-09-20 08:40:11.922 [warning] DiagnosticCollection with name 'apex-errors' does already exist.
2023-09-20 08:40:11.926 [info] ExtensionService#_doActivateExtension vscode.emmet, startup: false, activationEvent: 'onLanguage'
2023-09-20 08:40:11.944 [info] ExtensionService#_doActivateExtension vscode.git-base, startup: true, activationEvent: '*', root cause: vscode.git
2023-09-20 08:40:11.946 [info] ExtensionService#_doActivateExtension chuckjonas.apex-pmd, startup: true, activationEvent: '*'
2023-09-20 08:40:11.949 [info] ExtensionService#_doActivateExtension vscode.npm, startup: true, activationEvent: 'workspaceContains:package.json'
2023-09-20 08:40:11.984 [info] ExtensionService#_doActivateExtension salesforce.salesforcedx-vscode-core, startup: true, activationEvent: 'workspaceContains:sfdx-project.json', root cause: salesforce.salesforcedx-einstein-gpt
2023-09-20 08:40:14.019 [info] ExtensionService#_doActivateExtension salesforce.salesforcedx-vscode-lightning, startup: true, activationEvent: 'workspaceContains:sfdx-project.json'
2023-09-20 08:40:15.088 [info] ExtensionService#_doActivateExtension dbaeumer.vscode-eslint, startup: true, activationEvent: 'workspaceContains:sfdx-project.json', root cause: salesforce.salesforcedx-vscode-lwc
2023-09-20 08:40:15.650 [info] ExtensionService#_doActivateExtension salesforce.salesforcedx-vscode-soql, startup: true, activationEvent: 'workspaceContains:sfdx-project.json'
2023-09-20 08:40:16.666 [info] ExtensionService#_doActivateExtension vscode.github-authentication, startup: false, activationEvent: 'onAuthenticationRequest:github'
2023-09-20 08:40:16.673 [info] ExtensionService#_doActivateExtension vscode.git, startup: true, activationEvent: '*'
2023-09-20 08:40:16.688 [info] ExtensionService#_doActivateExtension vscode.github, startup: true, activationEvent: '*'
2023-09-20 08:40:16.695 [info] ExtensionService#_doActivateExtension salesforce.salesforcedx-vscode-lwc, startup: true, activationEvent: 'workspaceContains:sfdx-project.json'
2023-09-20 08:40:18.273 [info] ExtensionService#_doActivateExtension redhat.vscode-xml, startup: false, activationEvent: 'api', root cause: salesforce.salesforcedx-vscode-lwc
2023-09-20 08:40:18.433 [info] ExtensionService#_doActivateExtension vscode.microsoft-authentication, startup: false, activationEvent: 'onAuthenticationRequest:microsoft'
2023-09-20 08:40:19.855 [info] ExtensionService#_doActivateExtension salesforce.salesforcedx-einstein-gpt, startup: true, activationEvent: 'workspaceContains:sfdx-project.json'
2023-09-20 08:40:20.010 [info] ExtensionService#_doActivateExtension salesforce.salesforcedx-vscode-visualforce, startup: true, activationEvent: 'workspaceContains:sfdx-project.json'
2023-09-20 08:40:20.014 [info] ExtensionService#_doActivateExtension salesforce.salesforcedx-vscode-apex-debugger, startup: true, activationEvent: 'workspaceContains:sfdx-project.json'
2023-09-20 08:40:20.958 [info] ExtensionService#_doActivateExtension salesforce.salesforcedx-vscode-apex-replay-debugger, startup: true, activationEvent: 'workspaceContains:sfdx-project.json'
2023-09-20 08:40:22.252 [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch, startup: false, activationEvent: 'onStartupFinished'
2023-09-20 08:40:22.254 [info] ExtensionService#_doActivateExtension vscode.merge-conflict, startup: false, activationEvent: 'onStartupFinished'
2023-09-20 08:40:22.259 [info] ExtensionService#_doActivateExtension eamodio.gitlens, startup: false, activationEvent: 'onStartupFinished'
2023-09-20 08:40:22.283 [info] ExtensionService#_doActivateExtension esbenp.prettier-vscode, startup: false, activationEvent: 'onStartupFinished'
2023-09-20 08:40:22.301 [info] ExtensionService#_doActivateExtension GitHub.copilot, startup: false, activationEvent: 'onStartupFinished'
2023-09-20 08:40:22.338 [info] ExtensionService#_doActivateExtension GitHub.copilot-labs, startup: false, activationEvent: 'onStartupFinished'
2023-09-20 08:40:22.428 [info] ExtensionService#_doActivateExtension shardulm94.trailing-spaces, startup: false, activationEvent: 'onStartupFinished'
2023-09-20 08:40:22.476 [warning] [redhat.vscode-xml] Accessing a resource scoped configuration without providing a resource is not expected. To get the effective value for '[xml]', provide the URI of a resource or 'null' for any resource.
2023-09-20 08:40:22.480 [warning] [redhat.vscode-xml] Accessing a resource scoped configuration without providing a resource is not expected. To get the effective value for '[xml]', provide the URI of a resource or 'null' for any resource.
2023-09-20 08:40:22.488 [warning] [redhat.vscode-xml] Accessing a resource scoped configuration without providing a resource is not expected. To get the effective value for '[xml]', provide the URI of a resource or 'null' for any resource.
2023-09-20 08:40:23.417 [info] ExtensionService#_doActivateExtension ms-vsliveshare.vsliveshare, startup: false, activationEvent: 'api', root cause: eamodio.gitlens
2023-09-20 08:40:23.500 [error] Error: Request initialize failed with message: Cannot read properties of undefined (reading 'location')
    at be (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/dist/index.js:23:227)
    at Te (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/dist/index.js:22:12217)
    at Immediate.<anonymous> (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/dist/index.js:22:12068)
    at processImmediate (node:internal/timers:476:21)
2023-09-20 08:40:25.809 [info] Eager extensions activated
```

**Comment by pavan00089** [comment]
Even I am also experiencing the same issue.

![VS_Error](https://github.com/forcedotcom/salesforcedx-vscode/assets/104124504/57a6f8fc-22fe-473b-a0bd-7241d3e5f5a3)

**Comment by WuggyX2** [comment]
@CristiCanizales We are having this issue also. I was able to reproduce this issue in a freshly generated project. The extension crashes when there is another javascript file in a LWC component that references window.location. There might be other APIs that cause the language server to crash, not just window.location.

Steps to reproduce:
1. Generate a new sf project with cli.
2. Create new LWC  -component. Everything should work fine like referencing `window.localtion` 
3. Create a new JS into the lwc component created in the previous step.
4. create some code that references something in `window.location` like `window.location.href`. LWC extension should now crash.
5. Reload window -> language server crashes on launch.

Language server output
<img width="1230" alt="kuva" src="https://github.com/forcedotcom/salesforcedx-vscode/assets/51646225/0f750f47-4f83-425c-8fef-a2e87e691fed">

OS: macOs 13.5.2 (22G91)
extension version: v58.14.2

**Comment by bobalicious** [comment]
Following @WuggyX2 's post, commented out all references to `window.` and `location` in a codebase does not resolve the issue.  It may be a trigger for the error, but it appears to not be the only one.

Also, creating an **empty** additional JS file anywhere in the lwc folder causes the following:

```
/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/node_modules/@salesforce/lwc-language-server/lib/javascript/type-mapping.js:428
    const declarationLoc = externalToInternalLoc((_a = defaultExport === null || defaultExport === void 0 ? void 0 : defaultExport.location) !== null && _a !== void 0 ? _a : mainClassObj.location);
                                                                                                                                                                                           ^

TypeError: Cannot read properties of undefined (reading 'location')
    at mapLwcMetadataToInternal (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/node_modules/@salesforce/lwc-language-server/lib/javascript/type-mapping.js:428:188)
    at compileSource (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/node_modules/@salesforce/lwc-language-server/lib/javascript/compiler.js:172:66)
    at compileDocument (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/node_modules/@salesforce/lwc-language-server/lib/javascript/compiler.js:184:12)
    at Server.onDidChangeContent (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/node_modules/@salesforce/lwc-language-server/lib/lwc-server.js:195:84)`
```

**Comment by bobalicious** [comment]
Also, if I create a headless LWC I get an issue.

1. Start with empty project
2. Create LWC 'LayoutConstants'
3.  Replace JS contents with:

```
const DISPLAY_DENSITY = Object.freeze({
    COMFY   : 'comfy',
    COMPACT : 'compact',
    DEFAULT : 'comfy',
});

const getKeyForDisplayDensity = displayDensity => {
    for ( let key in DISPLAY_DENSITY ) {
        if ( DISPLAY_DENSITY[ key ] === displayDensity ) {
            return key;
        }
    }
	return '';
}

const LABEL_PROPERTIES = Object.freeze({
    COMFY   : {
        VARIANT : 'standard',
        CLASSES : 'slds-form-element slds-form-element_stacked'
    },
    COMPACT : {
        VARIANT : 'label-inline',
        CLASSES : 'slds-form-element slds-form-element_horizontal'
    },
    HIDDEN : {
        VARIANT : 'label-hidden',
        CLASSES : ''
    },
});

const CONSTANTS = Object.freeze({
    DISPLAY_DENSITY  : DISPLAY_DENSITY,
    LABEL_PROPERTIES : LABEL_PROPERTIES,
    getLabelVariant  : displayDensity => LABEL_PROPERTIES[ getKeyForDisplayDensity( displayDensity ) ].VARIANT,
    getLabelClasses  : displayDensity => LABEL_PROPERTIES[ getKeyForDisplayDensity( displayDensity ) ].CLASSES
});

export default CONSTANTS;
```

Immediate crash before save:


```
/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/node_modules/@salesforce/lwc-language-server/lib/javascript/type-mapping.js:373
    methodLocs, } = getDecoratedMethods(classObj.methods);
                                                 ^
TypeError: Cannot read properties of undefined (reading 'methods')
    at getDecorators (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/node_modules/@salesforce/lwc-language-server/lib/javascript/type-mapping.js:373:50)
    at mapLwcMetadataToInternal (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/node_modules/@salesforce/lwc-language-server/lib/javascript/type-mapping.js:430:21)
    at compileSource (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/node_modules/@salesforce/lwc-language-server/lib/javascript/compiler.js:172:66)
    at compileDocument (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/node_modules/@salesforce/lwc-language-server/lib/javascript/compiler.js:184:12)
    at Server.onDidChangeContent (/Users/xxxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-58.14.2/node_modules/@salesforce/lwc-language-server/lib/lwc-server.js:195:84)
```

I suspect that these are all related to the fact that the LWC language server is very strict about the shape of LWC it expects and doesn't handle LWCs that are actually modules, or have additional JS files in them.

**Comment by WuggyX2** [comment]
#4994 Error messages in this issue seem to be similar that we are having.

**Comment by gianlucaghezzi24** [comment]
I've updated the Lightning Web Components extension to version v58.15.0 but the error is still there

**Comment by CristiCanizales** [comment]
Hello everyone, could you please try installing the vsixes at the bottom of [this page](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/6329625411?pr=5126#:~:text=Size-,VS%20Code%20Extensions,-173%20MB), restart vscode and check if the issue persists? 

To install vsixes you need to go to the extensions view, click the 3 dots in the action menu, select Install from vsixes and select them:

![image](https://github.com/forcedotcom/salesforcedx-vscode/assets/113132642/99ae2971-30cb-461d-a3f5-ba3b45ecc94e)

**Comment by juanjetomas** [comment]
@CristiCanizales after updating the LWC extension from the VSIX the error stopped popping. I wasn't able to install all of them because there were some installation errors.

**Comment by CristiCanizales** [comment]
Hi @salesforceDev-role, @KevinLaurence, @JsalvaJ, @gianlucaghezzi24, @SalesforceAlexS, @rob-baillie-ortoo, @bobalicious, @pavan00089, @WuggyX2! Could you all please do the same (install VSIX for salesforce.salesforcedx-vscode-lwc extension which you can find at the bottom of [this page](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/6329625411?pr=5126#:~:text=Size-,VS%20Code%20Extensions,-173%20MB), restart vscode and check if the issue persists)? Thanks in advance!

**Comment by KevinLaurence** [comment]
> Hi @salesforceDev-role, @KevinLaurence, @JsalvaJ, @gianlucaghezzi24, @SalesforceAlexS, @rob-baillie-ortoo, @bobalicious, @pavan00089, @WuggyX2! Could you all please do the same (install VSIX for salesforce.salesforcedx-vscode-lwc extension which you can find at the bottom of [this page](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/6329625411?pr=5126#:~:text=Size-,VS%20Code%20Extensions,-173%20MB), restart vscode and check if the issue persists)? Thanks in advance!

I cannot download the file. Don't know why.

**Comment by salesforceDev-role** [comment]
Hello @CristiCanizales, I did try as you asked to download the mentioned extension file and install it. It seems to be working as expected, I am not getting that failed to initialize message anymore👍 . 
Happy Friday

**Comment by CristiCanizales** [comment]
Hello @salesforceDev-role, great! And yes, of course! It will go out in our next release (next wednesday/thursday). Thanks for your patience all this time and sorry for the inconveniences. Happy friday!

**Comment by CristiCanizales** [comment]
@KevinLaurence Here's the [.zip file for salesforcedx-vscode-lwc-58.15.0.zip](https://github.com/forcedotcom/salesforcedx-vscode/files/12770228/salesforcedx-vscode-lwc-58.15.0.zip). Just change the extension to .vsix and try installing it.

**Comment by salesforceDev-role** [comment]
@CristiCanizales, Thanks for all your help, appreciate it.

**Comment by WuggyX2** [comment]
@CristiCanizales I also tested the updated extension and can confirm that the language server no longer crashes when there are additional JS modules in a same file.

**Comment by CristiCanizales** [comment]
Closing this issue after confirmation from different users that the fix works. Thanks @divmain for the fix! 🥳

**Comment by pavan00089** [comment]
@CristiCanizales  Help me out here If I am doing something wrong.

1) I download the zip from the link then extracted it.

2) Then I renamed  extension.vsixmanifest file to  extension.vsixmanifest.vsix.

3) Then I tried to Install that manually but getting below error.

![Error](https://github.com/forcedotcom/salesforcedx-vscode/assets/104124504/e857ed72-5ec0-40e5-b2f3-c369962188fd)

**Comment by JsalvaJ** [comment]
Hello @CristiCanizales it seems that it works good. I don't have the issue when VS starts.
Thanks a lot!

**Comment by KevinLaurence** [comment]
Thanks @CristiCanizales I can confirm that the latest version of the extension installed in VS Code successfully.

**Comment by bobalicious** [comment]
@CristiCanizales - Sorry, I've not been able to install the upgrade linked from this issue - has this fix gone into the latest versions yet?  I see a new version was released yesterday.

If so, the problem still occurs for me.

**Comment by Razvan-Racolta** [comment]
I have the same issue. I downloaded the [.zip file for salesforcedx-vscode-lwc-58.15.0.zip](https://github.com/forcedotcom/salesforcedx-vscode/files/12770228/salesforcedx-vscode-lwc-58.15.0.zip), renamed the extension to vsix, installed it and after vs code restart, the error is still there. After extension update, the issue is still there.

Should this issue be reopened?

Thank you kindly @CristiCanizales

**Comment by JsalvaJ** [comment]
Hello, I'm continuing having the same issue. I installed the last version v59 and the same message is fired for me. 

mainThreadExtensionService.ts:81 [salesforce.salesforcedx-vscode-lwc]Request initialize failed with message: Cannot read properties of undefined (reading 'location')
$onExtensionRuntimeError @ mainThreadExtensionService.ts:81
mainThreadExtensionService.ts:82 Error: Request initialize failed with message: Cannot read properties of undefined (reading 'location')
	at be (\.vscode\extensions\salesforce.salesforcedx-vscode-lwc-59.0.0\dist\index.js:23:227)
	at Te (\.vscode\extensions\salesforce.salesforcedx-vscode-lwc-59.0.0\dist\index.js:22:12217)
	at Immediate.<anonymous> (\.vscode\extensions\salesforce.salesforcedx-vscode-lwc-59.0.0\dist\index.js:22:12068)
	at process.processImmediate (node:internal/timers:476:21)

**Comment by bobalicious** [comment]
@CristiCanizales - this ticket has had the label 'stale' added to it - and whilst the ticket may have had a low level replies from yourselves over the last week or so, it is still my understanding that it is active and should remain open.  Can the label be removed?

**Comment by CristiCanizales** [comment]
@JsalvaJ It was supposed to be solved but now it seems like it isn't. We're still investigating and will update once we have a new fix!

**Comment by gbockus-sf** [comment]
Howdy @bobalicious @JsalvaJ @Razvan-Racolta 
I built a debug version of the LWC extension with logging around when a location property is accessed.  If you could download from my fork and attempt to generate the debug logging to give us an idea of where the failure is occurring that would be awesome.  Find instructions and vsix in the below release. 
https://github.com/gbockus-sf/salesforcedx-vscode/releases/tag/0.0.3

I'm pretty sure the issue is something about how exports are structured, but haven't been able to repo based on the `LayoutConstants` example above.   

Thanks for your patience on this.

**Comment by bobalicious** [comment]
[lwc-language-server-log-23-10-19-0912.txt](https://github.com/forcedotcom/salesforcedx-vscode/files/13039814/lwc-language-server-log-23-10-19-0912.txt)

**Comment by bobalicious** [comment]
It'll be a few days before I'll be able to do that, but I will certainly try.

Knowing which LWC is causing the problem to appear, I may be able to reduce the test case and provide a version that is less, shall we say, private.  I'll catch up with this again next week.

**Comment by bobalicious** [comment]
### Reproduction using the new VSIX

With the version referenced above installed, I do not get an 'LWC Language Server' output window.

There is an 'LWC Extension' output window that contains no logs whatsoever.

I can see no other output  windows that contain any debug logs of the type you are looking for.

The 'Extension Host' output  window contains:

```
2023-10-30 11:10:04.500 [error] Error: Request initialize failed with message: Cannot read properties of undefined (reading 'location')
    at handleResponse (/Users/xxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-59.0.1/dist/index.js:8728:40)
    at processMessageQueue (/Users/xxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-59.0.1/dist/index.js:8583:13)
    at Immediate.<anonymous> (/Users/xxx/.vscode/extensions/salesforce.salesforcedx-vscode-lwc-59.0.1/dist/index.js:8569:11)
    at processImmediate (node:internal/timers:476:21)
```

### Replication using the LoadEntityForm LWC

I have attempted to replicate the issue using only the LoadEntityForm LWC and it does not occur with that LWC, and so I am not forwarding it on to you.

**Comment by bobalicious** [comment]
@gbockus-sf  / @CristiCanizales - I believe I've found the shape of LWC that causes the issue.

It is a pattern where you create a setter but no corresponding getter.

1. Create a new LWC
2. Add a property
3. Add a setter for that property with a slightly different name
4. DO NOT add a getter
5. Save
6. Witness the crash

E.g. an LWC that consists solely of the following:

```
import { LightningElement } from 'lwc';

export default class PropertySetterWithNoGetter extends LightningElement {

    _property = '';
	set property( value ) {
		this._property = value;
	}
}
```

Please do not discuss as to why this would be desirable - it is a perfectly valid block of code and I am aware that there are other ways of doing the same thing.  This example exists purely to show how to get the Lightning Web Components extension to crash.

**Comment by gbockus-sf** [comment]
💥 Thanks @bobalicious I repo'd locally and will see what we can do to get it resolved.

**Comment by gbockus-sf** [comment]
Fixed with the [59.3.1](https://github.com/forcedotcom/salesforcedx-vscode/releases/tag/v59.3.1) Release.  Thanks for working with us to get this issue resolved. 🥳 🎉

**Comment by bobalicious** [comment]
Confirmed as fixed from my perspective
